bigtable-hbase-1.x-parent/bigtable-hbase-1.x/src/test/java/com/google/cloud/bigtable/hbase1_x/TestMetrics.java

```diff
@@ -195,6 +236,147 @@ public void appendFailure() throws IOException {
     }
   }
 
+  @Test
+  public void testScanMetrics() throws IOException {
+    Scan scan = new Scan().withStartRow(rowKey).withStopRow(rowKey, true);
+    Table table = connection.getTable(TABLE_NAME);
+
+    Stopwatch stopwatch = Stopwatch.createStarted();
+    ResultScanner testScanner = table.getScanner(scan);
+    testScanner.next();
+    testScanner.close();
+    long methodInvocationLatency = stopwatch.elapsed(TimeUnit.MILLISECONDS);
+
+    fakeDataService.popLastRequest();
+
+    long scannerResultsLatencyMetric =
+        fakeMetricRegistry.results.get("google-cloud-bigtable.scanner.results.latency").get();
+    long scannerResultsMetric =
+        fakeMetricRegistry.results.get("google-cloud-bigtable.scanner.results").get();
+
+    assertThat(scannerResultsMetric).isEqualTo(1);
+    assertThat(scannerResultsLatencyMetric).isAtMost(methodInvocationLatency);
+  }
+
+  @Test
+  public void testFirstResponseLatency() throws IOException {
+    Scan scan = new Scan().withStartRow(rowKey).withStopRow(rowKey, true);
+    Table table = connection.getTable(TABLE_NAME);
+
+    Stopwatch stopwatch = Stopwatch.createStarted();
+    ResultScanner testScanner = table.getScanner(scan);
+    testScanner.next();
+    testScanner.close();
+    long methodInvocationLatency = stopwatch.elapsed(TimeUnit.MILLISECONDS);
+
+    fakeDataService.popLastRequest();
+
+    long firstResponseLatencyMetric =
+        fakeMetricRegistry
+            .results
+            .get("google-cloud-bigtable.grpc.method.ReadRows.firstResponse.latency")
+            .get();
+
+    // adding buffer time to the upper range to allow for a race between the emulator and the client
+    // recording the duration
+    assertThat(firstResponseLatencyMetric).isAtMost(methodInvocationLatency - 20 / 2);
+  }
+
+  @Test
+  public void testActiveSessionsAndChannels() throws IOException {
+    // There should already be 1 active session and 1 channel from connection (connecting to
+    // emulator will set pool size to 1)
+    long currentActiveSessions =
+        fakeMetricRegistry.results.get("google-cloud-bigtable.session.active").get();
+    long currentActiveChannels =
+        fakeMetricRegistry.results.get("google-cloud-bigtable.grpc.channel.active").get();
+
+    assertThat(currentActiveSessions).isEqualTo(1);
+    assertThat(currentActiveChannels).isEqualTo(1);
+
+    // Create a new session
+    int connectionCount = 10;
```

Comment:
can we extract some of this out with the next test's setup? I think the only difference is the credentials bit?